src/main/java/gov/usgs/wma/waterdata/groundwater/BuildRdbFile.java

```diff
@@ -85,6 +85,8 @@ protected ResultObject processRequest(String locationFolder) {
 			RdbWriter rdbWriter = createRdbWriter(writer).writeHeader();
 			dao.sendDiscreteGroundWater(states, rdbWriter);
 
+			s3bucket.sendS3();
+
```

Comment:
Now explicitly calling 'send' to prevent a potential bug - see below.

Reply:
well, now the impl is exposed to a higher level that did not need to know.

Reply:
I think the key issue here is the potential bug.  Suppose dao.`sendDiscreteGroundWater()` writes half the file, then throws an exception.  The try block / autoclose will still call close on the s3bucket and, with the current code, close will automatically send the partial file.  Sending a partial file is REALLY BAD because the behaviour on the receiving end is to delete everything for the site and replace it with the new - inadvertently deleting a bunch of data.

Reply:
I see. If we have an incomplete file due to an exception then the autoclose will send the incomplete file to s3. We need a mechanism to prevent the bad files from getting there. This impl is good.